src/reflect/scala/reflect/internal/pickling/UnPickler.scala

```diff
@@ -224,10 +224,9 @@ abstract class UnPickler {
           if (owner.isOverloaded)
             return NoSymbol
 
-          if (tag == EXTMODCLASSref) {
+          if (tag == EXTMODCLASSref && !owner.isInstanceOf[StubSymbol])
             owner.info.decl(nme.moduleVarName(name.toTermName))
-          }
-          NoSymbol
+          else NoSymbol
```

Comment:
Looks like it was only needed to be able to unpickle nested module references compiled before a5d47fb693d9b88ea9ed414762f16e027be64ada where the refered-to class was compiled under the new scheme. I'll remove the code.

https://issues.scala-lang.org/browse/SI-1591 and https://issues.scala-lang.org/browse/SI-3879.